### PR TITLE
Update Subdomain gateway section

### PR DESCRIPTION
**Pull request by lidel**
This PR aims to add better examples and clarify best practices around converting CID v0 to v1 for use in subdomains. 
It also mentions Base36 as a valid encoding in preparation for ED25519 libp2p-keys shipping with go-ipfs 0.7.0

cc https://github.com/ipfs/go-ipfs/pull/7441 @ipfs/wg-web-browsers

### CODE REVIEW

File: docs/how-to/address-ipfs-on-web.md [hunk 1]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 
-[PeerIDs can be represented as CID with `libp2p-key` multicodec](https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md):
+PeerIDs can be represented as [CID with `libp2p-key` multicodec](https://github.com/libp2p/specs/blob/master/RFC/0001-text-peerid-cid.md).
+Base36 is suggested as a safer default for longer keys:
 
 ```shell
-> ipfs cid format -v 1 -b base32 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
-bafzbeiagwnqiviaae5aet2zivwhhsorg75x2wka2pu55o7grr23ulx5kxm
+> ipfs cid format -v 1 -b base36 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
+k2k4r8jl0yz8qjgqbmc2cdu5hkqek5rj6flgnlkyywynci20j0iuyfuj
```
Comment (aschmahmann) [inline docs/how-to/address-ipfs-on-web.md:124]: This only works on RSA keys, Ed25519 keys fail here. Ed25519 keys aren't super widely used right now, but as go-ipfs is switching to using them by default and PeerIDs will still be encoded as base58 by default we might want to include.

`ipfs id --format "<id>" --peerid-base base36` or using `ipfs key list -l` (we unfortunately haven't added support yet for `ipfs key list -l mykey`, but hopefully that'll make it into either the final release or 0.8.0)

File: docs/how-to/address-ipfs-on-web.md [hunk 2]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 
-::: danger
+::: warning Known issues
+
+- Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+- DNS label length is limited to 63 characters ([RFC 1034](https://tools.ietf.org/html/rfc1034#page-7))
 
-Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
+Base32 is the safe default. Less popular Base36 can be used for longer ED25519 libp2p keys.
```
Comment (aschmahmann) [inline docs/how-to/address-ipfs-on-web.md:85]: I don't know what our end game is going to be around base representation for content CID (e.g. we may want to default to b36 everywhere for consistency), but this seems reasonable for now

File: docs/how-to/address-ipfs-on-web.md [hunk 3]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 :::
 
-::: danger
+::: warning Known issues
+
+- Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+- DNS label length is limited to 63 characters ([RFC 1034](https://tools.ietf.org/html/rfc1034#page-7))
 
-Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
```
Comment (aschmahmann) [inline docs/how-to/address-ipfs-on-web.md:84]: Are we advising people that they should canonically tell people to use `gateway.tld/path` and that it will redirect to the subdomain or is this unrelated?

File: docs/how-to/address-ipfs-on-web.md [hunk 4]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.dweb.link/wiki/
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.cf-ipfs.com/wiki/Vincent_van_Gogh.html
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.localhost:8080/wiki/
 
-::: tip
+::: tip Native support in go-ipfs 0.5+
 
-go-ipfs provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
+[go-ipfs](https://dist.ipfs.io/#go-ipfs) provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:70]:
```suggestion
[go-ipfs](https://dist.ipfs.io/#go-ipfs) provides native support for subdomain gateways on hostnames defined in the [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 5]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 - [`Gateway.PublicGateways` docs](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) for defining gateway behavior on specified hostnames
-- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) with ready to use one-liners for most common use cases
+- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gateway-recipes) with ready to use one-liners for most common use cases
 
 :::
 
-::: danger
+::: warning Known issues
+
+- Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:81]:
```suggestion
- Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before the HTTP gateway has a chance to read them
```

File: docs/how-to/address-ipfs-on-web.md (hunk 3 above)
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:84]:
```suggestion
Due to these limitations, the use of short, case-insensitive CIDv1 in a subdomain context is advised.
```

File: docs/how-to/address-ipfs-on-web.md (hunk 2 above)
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:85]:
```suggestion
Base32 is the safe default; the less-popular Base36 can be used for longer ED25519 libp2p keys.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 6]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+- DNS label length is limited to 63 characters ([RFC 1034](https://tools.ietf.org/html/rfc1034#page-7))
 
-Some browsers and other user agents force lowercase for the authority part of URLs, breaking case-sensitive CIDs before HTTP Gateway has a chance to read them.
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
+Base32 is the safe default. Less popular Base36 can be used for longer ED25519 libp2p keys.
 
-To avoid this, use of case-insensitive CIDv1 in Base32 in subdomain context is suggested as the safe default.
+See the next section to learn how to convert existing CID v0 to a DNS-safe representation.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:87]:
```suggestion
See the next section to learn how to convert an existing CIDv0 to a DNS-safe representation.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 7]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+    https://<cidv1b32>.ipfs.<gateway-host>.tld/path/to/resource
 
 Example:
 
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.dweb.link/wiki/
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.cf-ipfs.com/wiki/Vincent_van_Gogh.html
     https://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq.ipfs.localhost:8080/wiki/
 
-::: tip
+::: tip Native support in go-ipfs 0.5+
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:68]:
```suggestion
#### Native support in go-ipfs 0.5+
```
The sections marked out with `:::` turn into colored boxes in the rendered VuePress site, so we should use them sparingly.

File: docs/how-to/address-ipfs-on-web.md [hunk 8]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
-go-ipfs provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
+[go-ipfs](https://dist.ipfs.io/#go-ipfs) provides native support for subdomain gateways on hostnames defined in [`Gateway.PublicGateways`](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) configuration map.
 
 Learn more about daemon configuration for hosting a public gateway:
 
 - [`Gateway.PublicGateways` docs](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) for defining gateway behavior on specified hostnames
-- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gatewaypublicgateways) with ready to use one-liners for most common use cases
+- [`Gateway` recipes](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#gateway-recipes) with ready to use one-liners for most common use cases
 
 :::
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:77]:
```suggestion
```

File: docs/how-to/address-ipfs-on-web.md [hunk 9]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+Due to these limitaitons, use of short, case-insensitive CIDv1 in subdomain context is advised.
+Base32 is the safe default. Less popular Base36 can be used for longer ED25519 libp2p keys.
 
-To avoid this, use of case-insensitive CIDv1 in Base32 in subdomain context is suggested as the safe default.
+See the next section to learn how to convert existing CID v0 to a DNS-safe representation.
 
 :::
 
-::: tip
+::: tip CID conversion for subdomains
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:91]:
```suggestion
#### CID conversion for subdomains
```

File: docs/how-to/address-ipfs-on-web.md [hunk 10]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+Base36 is suggested as a safer default for longer keys:
 
 ```shell
-> ipfs cid format -v 1 -b base32 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
-bafzbeiagwnqiviaae5aet2zivwhhsorg75x2wka2pu55o7grr23ulx5kxm
+> ipfs cid format -v 1 -b base36 --codec libp2p-key QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN
+k2k4r8jl0yz8qjgqbmc2cdu5hkqek5rj6flgnlkyywynci20j0iuyfuj
 ```
 
 :::
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:127]:
```suggestion
```

File: docs/how-to/address-ipfs-on-web.md [hunk 11]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 
-To avoid this, use of case-insensitive CIDv1 in Base32 in subdomain context is suggested as the safe default.
+See the next section to learn how to convert existing CID v0 to a DNS-safe representation.
 
 :::
 
-::: tip
+::: tip CID conversion for subdomains
+
+If you have content identified by an older CID v0, there are easy ways to safely represent it as CID v1 for use in subdomains and other case-insensitive contexts.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:93]:
```suggestion
If you have content identified by an older CIDv0, there are easy ways to safely represent it as CIDv1 for use in subdomains and other case-insensitive contexts.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 12]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+#### Automatic – leverage the gateway in go-ipfs
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:95]:
```suggestion
#### Automatic — leverage the gateway in go-ipfs
```

File: docs/how-to/address-ipfs-on-web.md [hunk 13]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+**TL;DR:** using subdomain gateway as a drop-in replacement for a path one removes the need for manual CID conversion.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:97]:
```suggestion
**TL;DR:** Using a subdomain gateway as a drop-in replacement for a path one removes the need for manual CID conversion.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 14]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+Request for a content path sent to the gateway domain will return HTTP 301 redirect to a correct subdomain version, taking care of any necessary encoding conversion, if needed:
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:99]:
```suggestion
Request for a content path sent to the gateway domain will return an HTTP 301 redirect to a correct subdomain version, taking care of any necessary encoding conversion if needed:
```

File: docs/how-to/address-ipfs-on-web.md [hunk 15]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+    https://<gateway-host>.tld/ipfs/<cid> -> https://<cidv1>.ipfs.<gateway-host>.tld/
+
+To illustrate, opening the CID v0 resource at [`https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html`](https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html)
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:103]:
```suggestion
To illustrate, opening the CIDv0 resource at [`https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html`](https://dweb.link/ipfs/QmT5NvUtoM5nWFfrQdVrFtvGfKFmG7AHE8P34isapyhCxX/wiki/Mars.html)
```

File: docs/how-to/address-ipfs-on-web.md [hunk 16]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+returns a redirect to a CID v1 representation at [`https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html`](https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html).
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:104]:
```suggestion
returns a redirect to a CIDv1 representation at [`https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html`](https://bafybeicgmdpvw4duutrmdxl4a7gc52sxyuk7nz5gby77afwdteh3jc5bqa.ipfs.dweb.link/wiki/Mars.html).
```

File: docs/how-to/address-ipfs-on-web.md [hunk 17]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+Gateway takes care of converting CID to case-insensitive encoding.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:106]:
```suggestion
The gateway takes care of converting the CID to case-insensitive encoding.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 18]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+The multihash in CID v1 is the same as in original CID v0.
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:107]:
```suggestion
The multihash in CIDv1 is the same as in the original CIDv0.
```

File: docs/how-to/address-ipfs-on-web.md [hunk 19]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
+
+#### Manual – use CID website or commandline
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:109]:
```suggestion
#### Manual — use cid.ipfs.io or the command line
```

File: docs/how-to/address-ipfs-on-web.md [hunk 20]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
...
 
-To convert CID to Base32 use [cid.ipfs.io](https://cid.ipfs.io) or a command line:
+One can also do the conversion manually.
+To convert CID to Base32 ([RFC4648](https://tools.ietf.org/html/rfc4648#section-6), no padding) use [cid.ipfs.io](https://cid.ipfs.io) or a command line:
```
Comment (jessicaschilling) [inline docs/how-to/address-ipfs-on-web.md:112]:
```suggestion
To convert a CID to Base32 ([RFC4648](https://tools.ietf.org/html/rfc4648#section-6), no padding) use [cid.ipfs.io](https://cid.ipfs.io) or the command line:
```

File: docs/how-to/address-ipfs-on-web.md (hunk 3 above)
Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md]: Yes. I've included that in _CID conversion for subdomains_ section below. 
Should we make this more prominent somehow?

File: docs/how-to/address-ipfs-on-web.md (hunk 1 above)
Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md:123]: Eek. This edge case again :( 

Documenting this won't help much in practice: people will have pretty bad UX when `ipfs cid format` fails on ED25519 without an obvious reason, only a subset will read the docs and understand whats happening.

AFAIK `ipfs id` is the only place with non-CID PeerID when Ed25519 is used. 
`ipfs key list -l` prints Base36 version for `self` already.

@aschmahmann : how do you feel about `ipfs id` returning CIDv1b36 when PeerID is an identity multihash? It will close the UX gap.

I know its special-casing, but imo its worth it: will save us a lot of headache and time on user support in the following months.

Comment (aschmahmann) [reply docs/how-to/address-ipfs-on-web.md:123]: The issue is that the libp2p ecosystem is unprepared for peerIDs as CIDs and aside from this most recent go-libp2p (and go-ipfs) release nobody can process multiaddresses like `/p2p/k51abc...` which means the "normal" user flow of `ipfs swarm connect {copy-paste ipv4 part of multiaddress}/p2p/{copy-paste peerID} won't work on older nodes.

How commonly/why would people be converting the representations of peerIDs anyway? In my understanding using peerIDs in subdomains is not yet a commonly used feature. As a result my inclination is to just let this edge case be a bit miserable for a while and fix it later.

Some discussion on options around dealing with this are https://github.com/libp2p/go-libp2p/issues/976.

@Stebalien WDYT?

File: docs/how-to/address-ipfs-on-web.md (hunk 3 above)
Comment (aschmahmann) [reply docs/how-to/address-ipfs-on-web.md]: The question is more about what we're telling people. Do we tell them "hey, CIDs in subdomains is a security feature but the correct way for people to talk about URLs is using paths", or say "there are two valid ways to describe URLs"?

If it's the former then we should be louder/more prominent and explain that we'd like people to use CIDv1s so that things look nicer bc we redirect localhost:8080/ipfs/{cid} to {cid}.ipfs.localhost instead of to {cid in different representation}.ipfs.localhost. If it's the latter then we don't have to do anything.

What's your feeling on this?

File: docs/how-to/address-ipfs-on-web.md (hunk 1 above)
Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md:123]: Ack, I think p2p tunel is still experimental, so the only real life use will be IPNS on subdomains.
We already encourage people to publish to IPNS using a key other than `self`, so you are right, should be fine to leave it as-is.

Comment (aschmahmann) [reply docs/how-to/address-ipfs-on-web.md:123]: SGTM. Something that might help here is if we implement `ipfs key list -l {keyID}` since that would let us get the CID representation of the self key easily for publishing i.e. `ipfs key list -l self`.

Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md:123]: Good idea. I'll update this section with examples from https://github.com/ipfs/blog/pull/478

File: docs/how-to/address-ipfs-on-web.md [hunk 21]
```diff
@@ -55,49 +55,73 @@ https://ipfs.io/ipns/tr.wikipedia-on-ipfs.org/wiki/Anasayfa.html
 
 ### Subdomain gateway
 
-When [origin-based security](https://en.wikipedia.org/wiki/Same-origin_policy) perimeter is needed, [CIDv1](https://github.com/ipld/cid#cidv1) in Base32 ([RFC4648](https://tools.ietf.org/html/rfc4648#section-6), no padding) should be used in the subdomain:
+When [origin-based security](https://en.wikipedia.org/wiki/Same-origin_policy) perimeter is needed, [CIDv1](/concepts/content-addressing/#identifier-formats) in case-insensitive encoding such as Base32 or Base36 should be used in the subdomain:
```
[the remark this answers is not in the document]
Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md:58]: I've used "security perimeter" as in https://en.wikipedia.org/wiki/Perimeter_security, but if it sounds weird, we can drop "perimeter" and just have:

```suggestion
When [origin-based security](https://en.wikipedia.org/wiki/Same-origin_policy) is needed, [CIDv1](/concepts/content-addressing/#identifier-formats) in case-insensitive encoding such as Base32 or Base36 should be used in the subdomain:
```

Thoughts @johnnymatthews?

File: docs/how-to/address-ipfs-on-web.md (hunk 3 above)
Comment (lidel) [reply docs/how-to/address-ipfs-on-web.md]: @aschmahmann I think paths make it easier  to grasp the concept of "content addressing", so my intuition is that we should be thinking in therms of  "hey, CIDs in subdomains is a security feature but the correct way for people to talk about URLs is using paths". So `gateway.tld/path` remains "canonical". 

Not sure if we need to be more vocal: I believe people who care about subdomains already default to CIDv1, so the hash remains the same. What we cna do, is to ensure our GUIs copy path version instead of subdomain version etc.

Comment (aschmahmann) [reply docs/how-to/address-ipfs-on-web.md]: I think your approach is reasonable here as long as we're also trying to ensure that our comms focus on path domains. Sometimes subdomains will inevitably slip in as someone copy-pastes `cid.ipfs.localhost` from their URL bar and posts it somewhere, but not much to be done there at the moment.
